### PR TITLE
feat: support connecting Python Hive3 Namespace to a Kerberos-secured Hive Metastore

### PR DESCRIPTION
**Pull request by zouhuajian**
## Background

The current Python `Hive3Namespace` implementation only supports plain Thrift connections to Hive Metastore and cannot connect to Kerberized HMS deployments with `hive.metastore.sasl.enabled=true`.

## Changes

This PR adds Kerberos SASL support to the Python `Hive3Namespace` implementation while keeping the existing non-secure behavior unchanged.

Main changes include:

- add Kerberos SASL support to the Python Hive3 metastore wrapper
- use a Kerberos SASL transport when `hive.metastore.sasl.enabled=true`
- keep the existing plain Thrift path when SASL is not enabled
- support the following optional properties:
  - `hive.metastore.kerberos.principal`
  - `kerberos.service-name`
  - `kerberos.client-principal`
- add the required Python `hive3` dependencies
- add/update corresponding unit tests
- update the Hive3 documentation

## Configuration behavior

- `hive.metastore.sasl.enabled` is disabled by default
- `kerberos.service-name` takes precedence when explicitly configured
- otherwise it may be derived from `hive.metastore.kerberos.principal`
- `kerberos.client-principal` is optional; if omitted, local default Kerberos credentials are used

## Impact

This change enables the Python Hive3 Namespace to work with Kerberized Hive Metastore deployments without affecting existing non-SASL users.

## Testing

This PR includes test updates covering:

- default plain client behavior
- Kerberos configuration parsing
- Kerberos client selection when SASL is enabled
- plain client selection when SASL is disabled

## Notes

This change is scoped to Kerberos SASL support for the Python Hive3 Namespace only. It does not modify the Java Hive3 implementation and does not introduce a Kerberos integration test environment in this iteration.

---

Assisted by ChatGPT

### CODE REVIEW

File: python/src/lance_namespace_impls/hive3.py [hunk 1]
```diff
@@ -96,12 +145,264 @@
...
+                    )
+
+                if status not in (self.OK, self.COMPLETE):
+                    raise TTransportException(
+                        type=TTransportException.NOT_OPEN,
+                        message=f"Unexpected SASL negotiation status: {status}",
+                    )
+
+                if status == self.COMPLETE:
+                    break
```
[the remark this answers is not in the document]
Comment (zouhuajian) [reply python/src/lance_namespace_impls/hive3.py:260]: Thanks for the catch. We validated this path against a real Kerberized Hive Metastore, and in practice passing the `COMPLETE` payload back into `self._sasl.process(...)` causes the Python GSSAPI client to fail with an invalid-token error. With Hive Metastore, the handshake is already complete at that point, so we intentionally stop the negotiation on `COMPLETE` instead of feeding that payload back into puresasl.

I’ve also cleaned up the tests to make this behavior explicit: `OK` payloads are processed, empty `OK` payloads are normalized to `b""`, and `COMPLETE` ends the handshake even if it carries a non-empty payload.
